### ISSUE TITLE
Currently doesn't work on Typed Racket

### ISSUE THREAD

**Issue by takikawa** [racket/macro-debugger#7]
Running the macro stepper on this file does not work:

````
#lang typed/racket
"hello world"
````

I've included the debugging output here: https://gist.github.com/takikawa/df6328a95a4ccd3d568a

**Comment by samth** [comment]
This much smaller program has a derivation parse error as well:

```
#lang racket

(define-syntax (m stx)
  (syntax-case stx ()
    [(_ e) (local-expand #'e 'expression null)]))

(m 1)
```

**Comment by samth** [comment]
This program has a parse error that looks more like the TR one:

```
#lang racket
(begin-for-syntax (module* m racket))
```

The same program but with `module` instead of `module*` has a different derivation parse error.

**Comment by rmculpepper** [comment]
Fixed by cdffe710 and racket/racket@a41c63be.
